multimodal-search-pdf/pods/segmenters.py

```diff
@@ -0,0 +1,285 @@
+        It will then extract the data from the file, creating a list for images,
+        and text.
+
+        :param uri: File name of PDF
+        :type uri: str
+        :param buffer: PDF file in bytes
+        :type buffer: bytes
+        :param mime_type: the type of data
+        :returns: A list of documents with the extracted data
+        :rtype: List[Dict]
+        """
+        chunks=[]
+        if mime_type == 'application/pdf':
+            import fitz  # fitz is a library used in `PyMuPDF` to read pdf and images
+            import pdfplumber
+
+            if uri:
+                pdf_img = fitz.open(uri)
+                pdf_text = pdfplumber.open(uri)
+            elif buffer:
+                pdf_img = fitz.open(stream=buffer, filetype='pdf')
+                pdf_text_buffer = io.BytesIO(buffer)
+                pdf_text = pdfplumber.open(pdf_text_buffer)
+            else:
+                raise ValueError('No value found in "buffer" or "uri"')
+
+            chunks = []
+            # Extract images
+            with pdf_img:
+                for page in range(len(pdf_img)):
+                    for img in pdf_img.getPageImageList(page):
+                        xref = img[0]
+                        pix = fitz.Pixmap(pdf_img, xref)
+                        # read data from buffer and reshape the array into 3-d format
+                        np_arr = np.frombuffer(pix.samples, dtype=np.uint8).reshape(pix.h, pix.w, pix.n).astype('float32')
+                        if pix.n - pix.alpha < 4:  # if gray or RGB
+                            if pix.n == 1: #convert gray to rgb
+                                np_arr_rgb = np.concatenate((np_arr,)*3,-1)
+                                chunks.append(dict(blob=np_arr_rgb, weight=1.0, mime_type='image/png'))
+                            elif pix.n == 4: # remove transparency layer
+                                np_arr_rgb = np_arr[..., :3]
+                                chunks.append(dict(blob=np_arr_rgb, weight=1.0, mime_type='image/png'))
+                            else:
+                                chunks.append(dict(blob=np_arr, weight=1.0, mime_type='image/png'))
+                        else:  # if CMYK:
+                            pix = fitz.Pixmap(fitz.csRGB, pix)  # Convert to RGB
+                            np_arr = np.frombuffer(pix.samples, dtype=np.uint8).reshape(pix.h, pix.w, pix.n).astype(
+                                'float32')
+                            chunks.append(
+                                dict(blob=np_arr, weight=1.0, mime_type='image/png'))
+
+            # Extract text
+            with pdf_text:
+                count = len(pdf_text.pages)
+                for i in range(count):
+                    page = pdf_text.pages[i]
+                    text_page = page.extract_text(x_tolerance=1, y_tolerance=1)
+                    if text_page:
+                        chunks.append(dict(text=text_page, weight=1.0, mime_type='text/plain'))
```

Comment:
Duplicates with `L230`
```suggestion
            with pdf_text:
                count = len(pdf_text.pages)
                for i in range(count):
                    page = pdf_text.pages[i]
                    text_page = page.extract_text(x_tolerance=1, y_tolerance=1)
                    if text_page:
                        chunks.append(dict(text=text_page, weight=1.0, mime_type='text/plain'))
```

Reply:
The code of L230 is actually another class (MultimodalSegmenter), and the class is not used in the current flow. 
It's a combination of 3 segment functions(image,text and pdf). It can be used if we want to have all segmenters in 1 class. I'm not sure if I need to keep it or not.

Reply:
Rmoved